sampleExtension/background.js

```diff
@@ -0,0 +1,121 @@
+
+//Set that holds the urls to be intercepted
+var blockedSites = [];
+
+initExtension = function () {
+    //First receive the blacklist from the sync storage, and then create a onBeforeRequest listener using this list.
+    updateBlockedSites(replaceListener);
+    addBrowserActionListener();
+};
+
+// This function receives the blacklist from the sync storage.
+updateBlockedSites = function(callback){
+    chrome.storage.sync.get("tds_blacklist", function (items) {
+        if (!chrome.runtime.error) {
+            blockedSites = items.tds_blacklist;
+            return callback();
+        }
+    });
+};
+
+// This function adds one url to the blacklist
+addToBlockedSites = function(urlToAdd) {
+    var formatUrl = urlToAdd.split(['//'])[1];
+    formatUrl = '*://' + formatUrl + '*';
+    blockedSites.push(formatUrl);
+    chrome.storage.sync.set({"tds_blacklist" : blockedSites }, function() {
+        if(chrome.runtime.error) {
+            console.log("Runtime error.");
+        }
+    });
+};
+
+/* --------------- ------ Listener functions ------ ---------------*/
+
+replaceListener = function() {
+    chrome.webRequest.onBeforeRequest.removeListener(intercept);
+    addWebRequestListener();
+};
+
+addWebRequestListener = function() {
+    if(blockedSites != null && blockedSites.length > 0) {
+        chrome.webRequest.onBeforeRequest.addListener(
+                intercept,
+                {
+                    //Url's to be intercepted
+                    urls: blockedSites,
+                    types: ["main_frame"]
+            },
+            ["blocking"]
+        );
+    }
+};
+
+intercept = function() {
+    incrementInterceptionCounter();
+    return {redirectUrl: "https://zeeguu.herokuapp.com/getex"};
```

Comment:
What a pretty magical string, may be you should collect all the magical strings in their own file. I.e. no magic constants.